src/main/scala/org/ergoplatform/http/api/ErgoUtilsApiRoute.scala

```
@@ -60,9 +60,9 @@ class ErgoUtilsApiRoute(val ergoSettings: ErgoSettings)(
 
   def hashBlake2b: Route = {
     (post & path("hash" / "blake2b") & entity(as[Json])) { json =>
-      json.asString match {
-        case Some(message) => ApiResponse(encoder.encode(Blake2b256(message)))
-        case None          => ApiError.BadRequest
+      json.as[String] match {
+        case Right(message) => ApiResponse(encoder.encode(Blake2b256(message)))
+        case Left(ex)       => ApiError(StatusCodes.BadRequest, ex.getMessage())
```

It is better to propagate the error message to user